### PR TITLE
Update focus order in sankey chart for vertical navigation

### PR DESCRIPTION
**Pull request by krkshitij**
## Previous Behavior

The focus first moves through all the links and then the nodes, not following a logical sequence in the Sankey chart.

## New Behavior

The focus moves layer by layer in a vertical order. It first moves through the nodes in the 1st layer, then the links starting from these 1st layer nodes, and continues this pattern through each subsequent layer.

- Horizontal focus navigation has been disabled because the nodes and links are rendered vertically, causing the left/right arrow keys to move focus up or down to the previous or next sibling element.
- Bidirectional focus navigation has been disabled because it causes the up/down arrow keys to move the focus in a non-sequential and erratic manner within a 2D grid.

In FocusZone, the focus order is determined by the rendering order of the elements. Although the new focus order may not be entirely logical, it seems to be the most effective solution with FocusZone ensuring that the links (especially skip layer links) do not render over the nodes.

### Future Work

A more logical focus order in the Sankey chart would involve navigating focus horizontally, moving to the node first, then the connecting link, followed by the target node, and so on. This focus order needs to be implemented without altering the rendering order of nodes and links.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-charting-53e6e8eb-6b93-49b1-8419-7701781c49a0.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-charting-53e6e8eb-6b93-49b1-8419-7701781c49a0.json:1]: ## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31469&commitId=8cb073334dabcc70fe31c5114159f3e8187f85d4&env=prod&clientType=FLUENTUI)


  <details><summary><b>Pivot - Overflow</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Pivot - Overflow.Tabs - RTL.Narrow - Last tab selected.chromium.png|0| Removed|
|Pivot - Overflow.Tabs - RTL.Narrow - Overflow menu.chromium.png|0| Removed|

</details>


  <details><summary><b>react-charting-SankeyChart</b> 6 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-SankeyChart.Basic.default.chromium.png|683| Changed|
|react-charting-SankeyChart.PlaceHolder.default.chromium.png|234| Changed|
|react-charting-SankeyChart.PlaceHolder - RTL.default.chromium.png|234| Changed|
|react-charting-SankeyChart.Basic - Dark Mode.default.chromium.png|467| Changed|
|react-charting-SankeyChart.Basic - RTL.default.chromium.png|682| Changed|
|react-charting-SankeyChart.PlaceHolder - Dark Mode.default.chromium.png|191| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx [hunk 2]
```diff
@@ -759,6 +759,41 @@ export class SankeyChartBase extends React.Component<ISankeyChartProps, ISankeyC
         borderColorsForNodes,
       );
 
+      const nodeLinkDomOrderArray: { layer: number; type: string; index: number }[] = [];
+      nodes.sort((a: SNode, b: SNode) => {
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx:763]: Can you add in description what is fixed and what is left.

Comment (krkshitij) [reply packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx:763]: done

File: packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx [hunk 3]
```diff
@@ -792,12 +827,27 @@ export class SankeyChartBase extends React.Component<ISankeyChartProps, ISankeyC
...
+                if (item.type === 'node') {
+                  return (
+                    <g key={nodes[item.index].nodeId} className={classNames.nodes}>
+                      {nodeData![item.index]}
+                    </g>
+                  );
+                }
+
+                return (
+                  <g
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx:842]: add else corresponding to the if to make the code more readable

File: packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx [hunk 4]
```diff
@@ -792,12 +827,27 @@ export class SankeyChartBase extends React.Component<ISankeyChartProps, ISankeyC
...
+                    key={`${(links[item.index].source as SNode).nodeId}-${(links[item.index].target as SNode).nodeId}`}
+                    className={classNames.links}
+                    strokeOpacity={1}
+                  >
+                    {linkData![item.index]}
+                  </g>
+                );
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx:849]: Add the logic details that you have mentioned in the PR description as a comment in the code also.

**Review by AtishayMsft — APPROVED** [review]

File: packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx (hunk 3 above)
Comment (krkshitij) [reply packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx]: done

File: packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx (hunk 4 above)
Comment (krkshitij) [reply packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx]: done

**Pull request merged**